core/src/main/java/nu/yona/server/messaging/entities/BuddyMessage.java

```diff
@@ -111,17 +113,47 @@ public void decrypt()
 	 */
 	public static class BuddyInfoParameters
 	{
-		public final Optional<UUID> userPhotoId;
-		public final UUID userAnonymizedId;
+		public final UUID relatedUserAnonymizedId;
+
 		public final UUID userId;
 		public final String nickname;
+		public final Optional<UUID> userPhotoId;
 
-		public BuddyInfoParameters(UUID userId, UUID userAnonymizedId, String nickname, Optional<UUID> userPhotoId)
+		private BuddyInfoParameters(UUID relatedUserAnonymizedId, UUID userId, String nickname, Optional<UUID> userPhotoId)
 		{
+			this.relatedUserAnonymizedId = relatedUserAnonymizedId;
+
 			this.userId = userId;
-			this.userAnonymizedId = userAnonymizedId;
 			this.nickname = nickname;
 			this.userPhotoId = userPhotoId;
 		}
+
+		public static BuddyInfoParameters createInstance(UserDto userWithPrivateData)
```

Comment:
This creates a wrong dependency: the entities shouldn't be aware of DTOs. That's the reason why I proposed to put the ``createInstance...`` methods for ``BuddyInfoParameters`` on ``BuddyMessageDto``.